sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -332,6 +333,62 @@ private SpecializedBlobClientBuilder prepareBuilder() {
         }
     }
 
+
+    /**
+     * Creates a new block blob. By default this method will not overwrite an existing blob.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobAsyncClient.upload#BinaryData}
+     *
+     * @param data The data to write to the blob.
+     * @return A reactive response containing the information of the uploaded block blob.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<BlockBlobItem> upload(BinaryData data) {
+        try {
+            return upload(data, false);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Creates a new block blob, or updates the content of an existing block blob.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobAsyncClient.upload#BinaryData-boolean}
+     *
+     * @param data The data to write to the blob.
+     * @param overwrite Whether or not to overwrite, should the blob already exist.
+     * @return A reactive response containing the information of the uploaded block blob.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<BlockBlobItem> upload(BinaryData data,
```

Comment:
Do we want a `WithResponse` overload with BinaryData?

Reply:
That one is covered by `uploadWithResponse(BlobParallelUploadOptions options, Duration timeout, Context context)` and `uploadWithResponse(BlobParallelUploadOptions options)` with `BlobParallelUploadOptions(BinaryData)` ctor.